gtda/time_series/embedding.py

```diff
@@ -208,52 +213,169 @@ def plot(Xt, sample=0, plotly_params=None):
         return plot_point_cloud(Xt[sample], plotly_params=plotly_params)
 
 
+def time_delay_embedding(
+        X, time_delay=1, dimension=2, stride=1, flatten=False,
+        ensure_last_value=True, validate=True
+        ):
+    """Time-delay embeddings of arrays of time-series data.
+
+    On a 1D array `X` representing a single time series, the time-delay
+    embedding algorithm is the one described in :class:`TakensEmbedding` and
+    yields a 2D array representing a point cloud in Euclidean space. 2D array
+    inputs are interpreted as collections of time series (one per entry along
+    axis 0), and the algorithm is applied to each time series separately to
+    produce a 3D array of point clouds. More generally, N-dimensional array
+    inputs are interpreted as containing one time series per entry along the
+    first N - 1 axes, and a (N + 1)-dimensional array of point clouds is
+    returned.
+
+    Parameters
+    ----------
+    X : ndarray of shape (n_timestamps,) or (n_time_series, ..., n_timestamps)
+        Input time series or collection of time series.
+
+    time_delay : int, optional, default: ``1``
+        Time delay between two consecutive values for constructing one embedded
+        point.
+
+    dimension : int, optional, default: ``2``
+        Dimension of the embedding space.
+
+    stride : int, optional, default: ``1``
+        Stride duration between two consecutive embedded points.
+
+    flatten : bool, optional, default: ``False``
```

Comment:
Shouldn't it always returned a numpy array if the shapes of the outputs make sense?

Reply:
With `flatten=False`, `X_embedded` has 1 dimension more than `X` (whatever the dim of `X` is). With `flatten=True`, `X_embedded` has no more than 3 dimensions. Does this answer the question?